apis/atmos/src/main/java/org/jclouds/atmos/xml/ListDirectoryResponseHandler.java

```diff
@@ -49,11 +53,17 @@ public void endElement(String uri, String name, String qName) {
       } else if (qName.equals("FileType")) {
          currentType = FileType.fromValue(currentText.toString().trim());
       } else if (qName.equals("Filename")) {
+         currentFileName = currentText.toString().trim();
+         if (currentFileName.equals(""))
+            currentFileName = null;
+      } else if (qName.equals("Name")) {
          currentName = currentText.toString().trim();
```

Comment:
Previously `currentName` represented the file name.  Now `currentFileName` does this and we reuse `currentName` for individual metadata entries.  An abbreviated listing from the Atmos programmers guide:

```
<ListObjectsResponse xmlns='http://www.emc.com/cos/'>
<Object>
<ObjectID>499ad542a2a8bc200499ad5a7099940499b44f51e97d
</ObjectID>
<SystemMetadataList>
...
<Metadata>
<Name>size</Name>
<Value>7589</Value>
</Metadata>
...
</SystemMetadataList>
...
</SystemMetadataList>
```

Not listed, the Filename and FileType since the examples only have the object interface, not the namespace interface.